checkov/terraform/graph_builder/foreach/module_handler.py

```diff
@@ -75,8 +77,16 @@ def _duplicate_module_with_for_each(self, module_idx: int, for_each: dict[str, A
     def _duplicate_module_with_count(self, module_idx: int, count: int) -> None:
         self._create_new_resources_count(count, module_idx)
 
+    def _get_rendered_modules(self, source_modules: list[TFModule | None]) -> list[int]:
+        """
+        Returns a list of module indexes that have been rendered in the current iteration.
+        """
+        modules_created_by_provided_source_packed = [self.local_graph.vertices_by_module_dependency[curr][BlockType.MODULE] for curr in source_modules]
+        modules_created_by_provided_source = list(itertools.chain(*modules_created_by_provided_source_packed))  # list of lists -> single list
+        return modules_created_by_provided_source
+
     def _get_modules_to_render(self, current_level: list[TFModule | None]) -> list[int]:
-        rendered_modules = [self.local_graph.vertices_by_module_dependency[curr][BlockType.MODULE] for curr in current_level][0]
```

Comment:
Here we were pulling the first item from list of lists i.e `[[0]]`.
In nested modules use cases this array looked like `[[8], [10]]`, in such cases pulling `[0]` wasn't enough to render all required blocks